src/Security/Voter/AdminRestrictionVoter.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Security\Voter;
+
+use App\Entity\Participation;
+use App\Entity\Talk;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class AdminRestrictionVoter extends Voter
+{
+    protected function supports($attribute, $subject): bool
+    {
+        return \in_array($attribute, ['OWNER'])
+            && ($subject instanceof Talk || $subject instanceof Participation);
+    }
```

Comment:
I don't really like the naming here
* AdminRestrictionVoter
   *  Not need for `Restriction` here. There is already a `Voter` Suffix
   * I don't get what `Admin` means here. "On what are you voting" should be the root of the naming. Someting Like ParticipationVoter or TalkVoter is better. So yes you'll have to split the voter in 2. Anyway, there is too many `if` in the `voteOnAttribute` method.
* `OWNER` is not a good attribute name. It's better to use an action like `TALK_EDIT`, `TALK_SHOW`, `PARTICIPATION_EDIT` etc. And yes, you'll have to split the voter in two voters :) But it's almost always better to have "fine grained permissions"